src/java.base/share/classes/jdk/internal/classfile/constantpool/ClassEntry.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.constantpool;
+
+import java.lang.constant.ClassDesc;
+import java.lang.constant.ConstantDesc;
+import java.util.ArrayList;
+import java.util.List;
+
+import jdk.internal.classfile.impl.AbstractPoolEntry;
+import jdk.internal.classfile.impl.TemporaryConstantPool;
+import jdk.internal.classfile.impl.Util;
+
+
+/**
+ * Models a {@code CONSTANT_Class_info} constant in the constant pool of a
+ * classfile.
+ */
+public sealed interface ClassEntry
+        extends LoadableConstantEntry
+        permits AbstractPoolEntry.ClassEntryImpl {
+
+    @Override
+    default ConstantDesc constantValue() {
+        return asSymbol();
+    }
+
+    /**
+     * {@return the UTF8 constant pool entry for the class name}
+     */
+    Utf8Entry name();
+
+    /**
+     * {@return the class name, as an internal binary name}
+     */
+    String asInternalName();
+
+    /**
+     * {@return the class name, as a symbolic descriptor}
+     */
+    ClassDesc asSymbol();
+
+    /**
+     * Return a {@link List} composed by appending the additions to the base list.
+     * @param base The base elements for the list, must not include null
+     * @param additions The {@link ClassEntry} instances to add to the list, must not include null
+     * @return the combined {@link List}
+     */
+    static List<ClassEntry> adding(List<ClassEntry> base, List<ClassEntry> additions) {
```

Comment:
This and all the other following static methods seem more like implementation details rather than part of the public API?

Reply:
I've searched over all use cases we have and found no direct use of these API methods, so I've removed them.
We may later re-open discussion about possible API to combine and deduplicate lists of entries and symbols, however this isolated solution really does not fit and does not serve any purpose.